docs/snapshots.asciidoc

```diff
@@ -0,0 +1,190 @@
+
+Using ECK, you can automatically inject secure settings into a cluster node by providing them through a secret in the Elasticsearch Spec.
+
+. Create a file containing the GCS credentials. For this example, name it `gcs.client.default.credentials_file`. The file name is important as it will be reflected in the secure setting.
++
+[source,json]
+----
+{
+  "type": "service_account",
+  "project_id": "your-project-id",
+  "private_key_id": "...",
+  "private_key": "-----BEGIN PRIVATE KEY-----\n...\n-----END PRIVATE KEY-----\n",
+  "client_email": "service-account-for-your-repository@your-project-id.iam.gserviceaccount.com",
+  "client_id": "...",
+  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+  "token_uri": "https://accounts.google.com/o/oauth2/token",
+  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/your-bucket@your-project-id.iam.gserviceaccount.com"
+}
+----
+
+. Create a Kubernetes secret from that file:
++
+[source,sh]
+----
+kubectl create secret generic gcs-credentials --from-file=gcs.client.default.credentials_file
+----
+
+. Edit the `secureSettings` section of the Elasticsearch resource:
++
+[source,yaml]
+----
+kind: Elasticsearch
+spec:
+    # ...
+    # Inject secure settings into Elasticsearch nodes from a k8s secret reference
+    secureSettings:
+      secretName: "gcs-credentials"
+----
+
+. Apply the modifications:
++
+[source,bash]
+----
+kubectl apply -f elasticsearch.yml
+----
+
+GCS credentials are automatically propagated into each node's keystore. It can take up to a few minutes, depending on the number of secrets in the keystore. You don't have to restart the nodes.
+
+[id="{p}-create-repository"]
+== Register the repository in Elasticsearch
+
+. Create the GCS snapshot repository in Elasticsearch following the procedure described in https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-snapshots.html[Snapshot and Restore]:
++
+[source]
+----
+PUT /_snapshot/my_gcs_repository
+{
+  "type": "gcs",
+  "settings": {
+    "bucket": "my_bucket",
+    "client": "default"
+  }
+}
+----
+
+. Take a snapshot with the following HTTP request:
++
+[source]
+----
+PUT /_snapshot/my_gcs_repository/test-snapshot
+----
+
+[id="{p}-setup-cronjob"]
+== Periodic snapshots with a CronJob
+
+You can set up a simple CronJob to take a snapshot every day.
+
+. Make an HTTP request against the appropriate endpoint, using a daily snapshot naming format. Elasticsearch credentials are mounted as a volume into the job's pod:
++
+[source,yaml]
+----
+# snapshotter.yml
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: my-cluster-snapshotter
```

Comment:
The Elasticsearch is named `elasticsearch-sample` but this job `my-cluster`. I think I'd like to reuse the same name.